`valid-jsdoc` false positive: an object pattern in an assignment pattern.

**What version of ESLint are you using?**

master branch.

**What parser (default, Babel-ESLint, etc.) are you using?**

default.

**Please show your full configuration:**

nothing.

**What did you do? Please include the actual source code causing the issue.**

```js
/**
 * @param {string} a - a.
 * @param {object} [obj] - obj.
 * @param {string} obj.b - b.
 * @param {string} obj.c - c.
 * @returns {void}
 */
function foo(a, {b, c} = {}) {
    // empty
}
```

```
> eslint test.js --env es6 --rule valid-jsdoc:2 --no-ignore --no-eslintrc

C:\Users\starc\Documents\GitHub\eslint\test.js
  1:1  error  Expected JSDoc for 'undefined' but found 'obj'  valid-jsdoc

✖ 1 problem (1 error, 0 warnings)

```

**What did you expect to happen?**

No error.

**What actually happened? Please include the actual, raw output from ESLint.**

```
  1:1  error  Expected JSDoc for 'undefined' but found 'obj'  valid-jsdoc
```

--- a/lib/rules/valid-jsdoc.js
+++ b/lib/rules/valid-jsdoc.js
@@ -340,14 +340,14 @@ module.exports = {
 
                 if (node.params) {
                     node.params.forEach(function(param, i) {
-                        var name = param.name;
-
                         if (param.type === "AssignmentPattern") {
-                            name = param.left.name;
+                            param = param.left;
                         }
 
+                        var name = param.name;
+
                         // TODO(nzakas): Figure out logical things to do with destructured, default, rest params
-                        if (param.type === "Identifier" || param.type === "AssignmentPattern") {
+                        if (param.type === "Identifier") {
                             if (jsdocParams[i] && (name !== jsdocParams[i])) {
                                 context.report(jsdocNode, "Expected JSDoc for '{{name}}' but found '{{jsdocName}}'.", {
                                     name: name,

--- a/tests/lib/rules/valid-jsdoc.js
+++ b/tests/lib/rules/valid-jsdoc.js
@@ -512,6 +512,34 @@ ruleTester.run("valid-jsdoc", rule, {
             "*/\n" +
             "function foo(){}",
             options: [{ requireReturn: true }]
+        },
+        {
+            code: [
+                "/**",
+                " * @param {string} a - a.",
+                " * @param {object} [obj] - obj.",
+                " * @param {string} obj.b - b.",
+                " * @param {string} obj.c - c.",
+                " * @returns {void}",
+                " */",
+                "function foo(a, {b, c} = {}) {",
+                "    // empty",
+                "}"
+            ].join("\n"),
+            parserOptions: {ecmaVersion: 6}
+        },
+        {
+            code: [
+                "/**",
+                " * @param {string} a - a.",
+                " * @param {any[]} [list] - list.",
+                " * @returns {void}",
+                " */",
+                "function foo(a, [b, c] = []) {",
+                "    // empty",
+                "}"
+            ].join("\n"),
+            parserOptions: {ecmaVersion: 6}
         }
     ],
 

